sklearn/neighbors/tests/test_ball_tree.py

```diff
@@ -84,3 +85,13 @@ def test_array_object_type():
     X = np.array([(1, 2, 3), (2, 5), (5, 5, 1, 2)], dtype=object)
     with pytest.raises(ValueError, match="setting an array element with a sequence"):
         BallTree(X)
+
+
+def test_bad_pyfunc_metric():
+    def wrong_distance(x, y):
+        return "1"
+
+    X = np.ones((5, 2))
+    msg = "Custom distance function must accept two vectors"
+    with pytest.raises(TypeError, match=msg):
+        BallTree(X, metric=wrong_distance)
```

Comment:
This seems to be unrelated but I am fine with adding a new test if existing test did not cover this case.

Reply:
Oops, this test is not supposed to come here, I duplicated it there by inadvertence.